The code in spamass-milter.cpp that tries to create a
Sendmail-compatible header is broken and generates a header
that is incorrectly parsed by SpamAssassin.

This is mostly fixed in CVS apart from the space that needs
adding prior to the "(" between macro_j and macro_v.

https://savannah.nongnu.org/bugs/index.php?17178
http://bugs.debian.org/510665
http://bugzilla.redhat.com/496763

--- spamass-milter-0.3.1/spamass-milter.cpp.ori	2006-03-23 15:41:36.000000000 -0600
+++ spamass-milter-0.3.1/spamass-milter.cpp	2009-04-20 20:03:31.000000000 -0500
@@ -1002,9 +1002,9 @@
 
 		assassin->output((string)
 			"Received: from "+macro_s+" ("+macro__+")\r\n\t"+
-			"by "+macro_j+"("+macro_v+"/"+macro_Z+") with "+macro_r+" id "+macro_i+"\r\n\t"+
+			"by "+macro_j+" ("+macro_v+"/"+macro_Z+") with "+macro_r+" id "+macro_i+";\r\n\t"+
 			macro_b+"\r\n\t"+
-			"(envelope-from "+assassin->from()+"\r\n");
+			"(envelope-from "+assassin->from()+")\r\n");
 
 	} else
 		assassin->output((string)"X-Envelope-To: "+envrcpt[0]+"\r\n");
